### DISCUSSION TITLE
How to wait for user to load before rendering?

### DISCUSSION THREAD

**Discussion by sarkarology** [supabase/supabase#6378] [Questions]
I am using Supabase auth (with NextJS). How do I wait for the user to load before rendering the page?

Currently a logged out page is flashing before the usersession is loaded and then the logged in page loads.

How do I prevent that flash till the user session is getting loaded?

--

More specifically, when a user signs in using google, and gets redirected to the home page, the home page first loads the logged-out status, and after a second loads the logged-in session data. But for that one second the user's status is "logged out".

**Comment by ghost** [comment]
After searching a lot. I seen that you cannot use supabase auth for server side rendering from _app.js in next.js.

You implement Instagram-like logic where the loading screen appears while logging in.


This is my code
https://gist.github.com/studywithdesign/7f8abd32739c5b9a45b70790a058b39f

**Reply by AurelianSpodarec** [reply]
No code :(
